src/test/java/org/openrewrite/java/testing/jmockit/JMockitMockUpToMockitoTest.java

```diff
@@ -0,0 +1,379 @@
+                                                return 1024;
+                                            }
+                                
+                                            @Mock
+                                            public int staticMethod(int v) {
+                                                return 128;
+                                            }
+                                        };
+                                
+                                        // Should ignore the newClass statement
+                                        new Runnable() {
+                                            @Override
+                                            public void run() {
+                                                System.out.println("run");
+                                            }
+                                        };
+                                        assertEquals("newMsg", new MyClazz().getMsg());
+                                        assertEquals(1024, MyClazz.staticMethod());
+                                        assertEquals(128, MyClazz.staticMethod(0));
+                                    }
+                                
+                                    public static class MyClazz {
+                                        public String getMsg() {
+                                            return "msg";
+                                        }
+                                
+                                        public static int staticMethod() {
+                                            return 0;
+                                        }
+                                
+                                        public static int staticMethod(int v) {
+                                            return 1;
+                                        }
+                                    }
+                                }
+                                """,
+                        """
+                                import org.junit.Test;
+                                import org.mockito.MockedConstruction;
+                                import org.mockito.MockedStatic;
+                                
+                                import static org.junit.Assert.assertEquals;
+                                import static org.mockito.Mockito.*;
+                                
+                                public class MockUpTest {
+                                    @Test
+                                    public void test() {
+                                        final String msg = "newMsg";
+                                        MockedConstruction<MockUpTest.MyClazz> mockObjMockUpTest_MyClazz = mockConstruction(MockUpTest.MyClazz.class, (mock, context) -> {
+                                            when(mock.getMsg()).thenAnswer(invocation -> {
+                                                return msg;
```

Comment:
@timtebeek Pls have a look here. This test case is totally functional. But openrewrite tells "/*~~(Identifier type is missing or malformed)~~>*/msg;". Do you think I should change the test case?

Reply:
@shivanisky how do you think about this rewrite?

Reply:
I might not get to this till the weekend, and weekend is looking full, but will try to have a quick look. I think for mockup, need to just create a spy object and then add a when clause for that method? 

Not super versed on mockup - have only seen it 2x so far, but can try to help as I had to migrate them. One mockup case didn't get migrated as mockito has no possible option for that.

Reply:
I don't think Spy is a substitution of MockUp, since it may includes static method.

Actually my question here is that OpenRewrite can't detect the type of `msg` in line 335, which is defined in line 332, and it causes UT fail.

Reply:
I'd also need a bit of a closer look; I think the issue here is that the statement refers to a field that is only introduced later, meaning it's not available to the JavaTemplate. You can see that by adding `.doBeforeParseTemplate(System.out::println)` to your JavaTemplate builder. The printed class will contain the `msg` as used, but likely not it's definition, leading to missing type attribution.

I guess you'd worked around that previously by printing the statements and making them part of the template generated. 🤔